scipy/spatial/distance.py

```diff
@@ -1217,20 +1217,22 @@ def pdist(X, metric='euclidean', p=2, w=None, V=None, VI=None):
         if metric == minkowski:
             def dfun(u, v):
                 return minkowski(u, v, p)
+            X = _convert_to_double(X)
```

Comment:
Either converting to double is needed for all built-in distance functions or none of them I'd think. Since most are going through the ``else`` clause here, this looks inconsistent.